CMakeLists.txt

```diff
@@ -164,3 +228,191 @@ ExternalProject_Add(grpc
+    -DCMAKE_INSTALL_PREFIX:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/crc32c
+)
+
+# This is necessary because google-cloud-cpp
+# does not pass protobuf_DIR and gRPC_DIR etc to its
+# external projects but expects them in CMAKE_PREFIX_PATH.
+#
+# absl must preceed grpc to ensure that the absl from the newer grpc
+# repo is used and not the absl from the older grpc.
+set(GCS_CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH}
+   ${TRITON_THIRD_PARTY_INSTALL_PREFIX}/c-ares/lib/cmake/c-ares
+   ${TRITON_THIRD_PARTY_INSTALL_PREFIX}/absl/lib/cmake/absl
+   ${TRITON_THIRD_PARTY_INSTALL_PREFIX}/crc32c/lib/cmake/Crc32c
+   ${TRITON_THIRD_PARTY_INSTALL_PREFIX}/grpc/lib/cmake/grpc
+   ${TRITON_THIRD_PARTY_INSTALL_PREFIX}/nlohmann_json/lib/cmake/nlohmann_json
+   ${_FINDPACKAGE_PROTOBUF_CONFIG_DIR})
+
+#
+# Build google-cloud-cpp
+#
+ExternalProject_Add(google-cloud-cpp
+  PREFIX google-cloud-cpp
+  GIT_REPOSITORY "https://github.com/googleapis/google-cloud-cpp.git"
+  GIT_TAG "v1.22.0"
+  SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/google-cloud-cpp/src/google-cloud-cpp"
+  CMAKE_CACHE_ARGS
+    ${_CMAKE_ARGS_CMAKE_TOOLCHAIN_FILE}
+    ${_CMAKE_ARGS_VCPKG_TARGET_TRIPLET}
+    -DGOOGLE_CLOUD_CPP_DEPENDENCY_PROVIDER:STRING=package
+    -DBUILD_TESTING:BOOL=OFF
+    -DCMAKE_PREFIX_PATH:PATH=${GCS_CMAKE_PREFIX_PATH}
+    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+    -DCMAKE_INSTALL_PREFIX:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/google-cloud-cpp
+    -Dnlohmann_json_DIR:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/nlohmann_json/lib/cmake/nlohmann_json
+    -DProtobuf_DIR:PATH=${_FINDPACKAGE_PROTOBUF_CONFIG_DIR}
+    -DCrc32c_DIR:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/crc32c/lib/cmake/Crc32c
+  DEPENDS grpc c-ares protobuf crc32c absl nlohmann-json
+)
+
+#
+# Build azure-storage-cpplite
+#
+ExternalProject_Add(azure-storage-cpplite
+  PREFIX azure-storage-cpplite
+  GIT_REPOSITORY "https://github.com/Azure/azure-storage-cpplite.git"
+  GIT_TAG "v0.3.0"
+  SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/azure-storage-cpplite/src/azure-storage-cpplite"
+  CMAKE_CACHE_ARGS
+    ${_CMAKE_ARGS_CMAKE_TOOLCHAIN_FILE}
+    ${_CMAKE_ARGS_VCPKG_TARGET_TRIPLET}
+    -DBUILD_SHARED_LIBS:BOOL=OFF
+    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+    -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+    -DCURL_INCLUDE_DIR:STRING=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/curl/include
+    -DCURL_LIBRARY:STRING=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/curl/lib
+    -DCMAKE_INSTALL_PREFIX:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/azure-storage-cpplite
+  DEPENDS curl
+)
+
+#
+# Build CNMeM (CUDA memeory management library)
+#
+ExternalProject_Add(cnmem
+  PREFIX cnmem
+  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cnmem/cnmem"
+  DOWNLOAD_COMMAND ""
+  CMAKE_CACHE_ARGS
+    ${_CMAKE_ARGS_CMAKE_TOOLCHAIN_FILE}
+    ${_CMAKE_ARGS_VCPKG_TARGET_TRIPLET}
+    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+    -DCMAKE_INSTALL_PREFIX:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/cnmem
+)
+
+#
+# Build AWS sdk for S3 support
```

Comment:
To make this symmetric with gcp and azure we can choose between the two formats to stick to.
```
# Build google-cloud-cpp
```
and 
```
# Build AWS sdk for S3 support
```

Reply:
I didn't see this till now. I will fix in a separate change.